### DISCUSSION TITLE
can someone explain how depth works?

### DISCUSSION THREAD

**Discussion by yuen07** [zricethezav/gitleaks#570] [Q&A]
How can i just scan the master "branch" without scanning all the commits? i found that my report comes with 20k leaks but most of the leaks are due to repeating problems but in different commits, but they all share the same file and problem.

**Comment by w0rmr1d3r** [comment]
Hello @yuen07 ,
As you can see in this section -> https://github.com/zricethezav/gitleaks#usage-and-options
You can use the `--no-git` option.

Cheers!

**Reply by yuen07** [reply]
Thanks! i will try again with --no-git 

i was using the --depth parameter but i dont understand the usage

**Reply by zricethezav** [reply]
@yuen07 `--depth` will scan to the nth commit moving backwards from the default HEAD. Gitleaks will stop scanning at the nth commit.
